File: rb/features.rb
# NebulumMarsRovers SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module NebulumMarsRoversFeatures
  def self.make_feature(name)
    case name
    when "base"
      NebulumMarsRoversBaseFeature.new
    when "ratelimit"
      NebulumMarsRoversRatelimitFeature.new
    when "retry"
      NebulumMarsRoversRetryFeature.new
    when "test"
      NebulumMarsRoversTestFeature.new
    when "timeout"
      NebulumMarsRoversTimeoutFeature.new
    else
      NebulumMarsRoversBaseFeature.new
    end
  end
end

File: py/features.py
# NebulumMarsRovers SDK feature factory

from feature.base_feature import NebulumMarsRoversBaseFeature
from feature.test_feature import NebulumMarsRoversTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NebulumMarsRoversBaseFeature(),
        "test": lambda: NebulumMarsRoversTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
